feat(customer-domains) Prototype of URL swapping

markstory opened the pull request:
This is a quick & dirty solution that replaces organization slugs with slugless routes. This behavior would be enabled for customers with domains enabled and will help us get existing customers onto new paths while we straddle customer domains + path slugs.

Refs HC-481

markstory committed:
feat(customer-domains) Prototype of URL swapping

This is a quick & dirty solution that replaces organization slugs with
slugless routes. This behavior would be enabled for customers with
domains enabled and will help us get existing customers onto new paths
while we straddle customer domains + path slugs.

Refs HC-481

markstory committed:
Merge branch 'master' into feat-path-translations

markstory committed:
Add more scenarios and special routes.

markstory committed:
Handle path swap required for #41425

markstory committed:
Include trailing segments

markstory committed:
Merge branch 'master' into feat-path-translations

markstory committed:
initialData isn't always defined.

markstory committed:
Normalize URLs generated by settings as well.

markstory committed:
Fix more settings routes.

vbro reviewed (approved):
LGTM
Is there backend version of this for APIs?

markstory committed:
Merge branch 'master' into feat-path-translations

markstory committed:
Fix URL generation in issue list tabs

markstory committed:
Fix URL generation for tabs in performance summary.

markstory committed:
Normalize URLs in list links too

markstory committed:
Fix account URL generation

markstory committed:
Can't normalize in replaceRouterParams

We often generate URLs with recreateRoute() and then feed the result
into a Link component. This results in double normalization which is
bad.

Move normalization into navigateTo() and replace usage of
recreateRoute() with string URL generation which should be more
performant and work better with our URL swapping solution.

markstory committed:
Add more test scenarios

markstory committed:
Update settings views to use string URLs that are easier to normalize for customer domains

markstory committed:
Fix switching orgs via bread crumbs.
